compiler/rustc_hir_analysis/messages.ftl

```diff
@@ -57,6 +66,23 @@ hir_analysis_copy_impl_on_type_with_dtor =
     the trait `Copy` cannot be implemented for this type; the type has a destructor
     .label = `Copy` not allowed on types with destructors
 
+hir_analysis_dispatch_from_dyn_coercion = the trait `DispatchFromDyn` may only be implemented for a coercion between structures with the same definition; expected `{$source_path}`, found `{$target_path}`
+
+hir_analysis_dispatch_from_dyn_multi = implementing the `DispatchFromDyn` trait requires multiple coercions
+    .note = the trait `DispatchFromDyn` may only be implemented for a coercion between structures with a single field being coerced
+    .coercions_note = currently, {$number} fields need coercions: {$coercions}
+
+hir_analysis_dispatch_from_dyn_repr = structs implementing `DispatchFromDyn` may not have `#[repr(packed)]` or `#[repr(C)]`
+
+hir_analysis_dispatch_from_dyn_same = the trait `CoerceUnsized` may only be implemented for a coercion between structures with the same definition; expected `{$source_path}`, found `{$target_path}`
+
+hir_analysis_dispatch_from_dyn_single = the trait `DispatchFromDyn` may only be implemented for a coercion between structures with a single field being coerced, none found
+
+hir_analysis_dispatch_from_dyn_struct = the trait `DispatchFromDyn` may only be implemented for a coercion between structures
```

Comment:
This is the same as `hir_analysis_coerce_unsized_may`. Can you deduplicate it by making the trait name as an input parameter?